### PR TITLE
chromium: update to 86.0.4240.111

### LINKED ISSUE #417: Chromium: update to 86.0.4240.111
Stable release announcements since our latest update:
https://chromereleases.googleblog.com/2020/10/stable-channel-update-for-desktop_20.html

Targets and platforms we need to check:

**chromium-x11:**
- zeus, clang, MACHINE="intel-core2-32"
- zeus, clang, MACHINE="intel-corei7-64"
- zeus, clang, MACHINE="raspberrypi3"
- dunfell, MACHINE="raspberrypi3"

**chromium-ozone-wayland:**

- zeus, clang, MACHINE="m3ulcb"
- zeus, clang, MACHINE="qemuarm64"
- dunfell, clang, MACHINE="raspberrypi3"
- dunfell, clang, MACHINE="m3ulcb"

### PR DESCRIPTION
**Pull request by msisov**
Stable release announcements since our latest update:
https://chromereleases.googleblog.com/2020/10/stable-channel-update-for-desktop_20.html

Test-built with:

* chromium-x11
  * zeus, MACHINE="intel-core2-32"
  * zeus, MACHINE="intel-corei7-64"
  * zeus, MACHINE="raspberrypi3"
  * dunfell, MACHINE="raspberrypi3"

* chromium-ozone-wayland:
  * zeus, MACHINE="qemuarm64"
  * zeus, MACHINE="m3ulcb"
  * dunfell, MACHINE="raspberrypi3"
  * dunfell, MACHINE="m3ulcb"

License changes
---------------
Updated licenses:
 * third_party/libjpeg_turbo/LICENSE.md
   - Roll PDFium from 1553832ddd6c to 61ca10cb63d0 (5 revisions)
   - https://chromium-review.googlesource.com/c/chromium/src/+/2350136

Added licenses:
 * third_party/angle/src/libANGLE/renderer/vulkan/shaders/src/third_party/ffx_spd/LICENSE
   - Vulkan: Generate mipmap in compute
   - https://chromium-review.googlesource.com/c/angle/angle/+/2278713
 * third_party/nearby/LICENSE
   - Import nearby connections library.
   - https://chromium-review.googlesource.com/c/chromium/src/+/2266881
 * third_party/securemessage/LICENSE
   third_party/ukey2/LICENS
   - Import securemessage and ukey2 as third party libraries in chromium
   - https://chromium-review.googlesource.com/c/chromium/src/+/2274625
 * third_party/text-fragments-polyfill/LICENSE
   - Adding text-fragments-polyfill to //third-party
   - https://chromium-review.googlesource.com/c/chromium/src/+/2332840
 * third_party/tint/LICENSE
   - Add Tint to third_party/
   - https://chromium-review.googlesource.com/c/chromium/src/+/2248753
 * third_party/zxcvbn-cpp/LICENSE.txt
   - [third_party] Add zxcvbn-cpp
   - https://chromium-review.googlesource.com/c/chromium/src/+/2167393

Removed licenses:
 * third_party/cacheinvalidation/src/google/cacheinvalidation/COPYING
   - Delete third_party/cacheinvalidatio
   - https://chromium-review.googlesource.com/c/chromium/src/+/2274884

Android-only:
Added licenses:
 * third_party/android_deps/libs/androidx_exifinterface_exifinterface/LICENSE
   - Add's Androidx ExifInterface as a third-party dependency
   - https://chromium-review.googlesource.com/c/chromium/src/+/2283413
 * third_party/android_deps/libs/androidx_slice_slice_builders/LICENSE
   third_party/android_deps/libs/androidx_slice_slice_core/LICENSE
   - [Chromecast] Add androidx slices deps.
   - https://chromium-review.googlesource.com/c/chromium/src/+/2310872
 * third_party/android_deps/libs/androidx_webkit_webkit/LICENSE
   - Add androidx.webkit to third_party
   - https://chromium-review.googlesource.com/c/chromium/src/+/2264441
 * third_party/android_deps/libs/androidx_window_window/LICENSE
   - Adding androidx.window to android_deps
   - https://chromium-review.googlesource.com/c/chromium/src/+/2354599
 * third_party/android_deps/libs/com_android_tools_desugar_jdk_libs_configuration/LICENSE
   - Adds Desugar's jdk_libs_configuration to the build
   - https://chromium-review.googlesource.com/c/chromium/src/+/2354518

Fixes #417

### CODE REVIEW

File: recipes-browser/chromium/files/0001-Remove-dead-reloc-in-nonalloc-LD-flags.patch [hunk 1]
```diff
@@ -0,0 +1,32 @@
+From e5f288d4807b63381e497475de316926479ec427 Mon Sep 17 00:00:00 2001
+From: Maksim Sisov <msisov@igalia.com>
+Date: Mon, 26 Oct 2020 11:06:56 +0200
+Subject: [PATCH] Remove dead-reloc-in-nonalloc LD flags
```
Comment (rakuco) [inline recipes-browser/chromium/files/0001-Remove-dead-reloc-in-nonalloc-LD-flags.patch:4]: Can you provide more context to this patch (and the "Upstream-status" header on top)? I don't understand what problem is being solved here.

File: recipes-browser/chromium/files/0001-add-missing-cstring-include.patch [hunk 2]
```diff
@@ -0,0 +1,31 @@
+From 1da727488fc60b663f9afdc2ad033d5f0edc05fe Mon Sep 17 00:00:00 2001
```
Comment (rakuco) [inline recipes-browser/chromium/files/0001-add-missing-cstring-include.patch:1]: You're missing the Upstream-status + signed-off-by header.

**Review by rakuco — COMMENTED** [review]
Mostly lgtm, but there are a few issues with the new patches. One other thing: #416 had a change to `conf/layers.conf` that is not present here, and which has not been sent separately either. What's the status of that?

File: recipes-browser/chromium/files/0001-Remove-dead-reloc-in-nonalloc-LD-flags.patch (hunk 1 above)
Comment (msisov) [reply recipes-browser/chromium/files/0001-Remove-dead-reloc-in-nonalloc-LD-flags.patch:4]: Oh, sorry. I forgot about this one.

**Comment by r1mikey** [comment]
Tested on x86_64, Dunfell in Kiosk mode.

Tested functionality:
* Extensions
* Nativehost
* Touch
* GL acceleration

**Comment by rakuco** [comment]
Thanks for checking, @r1mikey!

File: recipes-browser/chromium/files/0001-Remove-dead-reloc-in-nonalloc-LD-flags.patch (hunk 1 above)
Comment (msisov) [reply recipes-browser/chromium/files/0001-Remove-dead-reloc-in-nonalloc-LD-flags.patch:8]: I've added a new comment explaining that.

File: recipes-browser/chromium/files/0001-add-missing-cstring-include.patch (hunk 2 above)
Comment (msisov) [reply recipes-browser/chromium/files/0001-add-missing-cstring-include.patch:5]: done

**Comment by rakuco** [comment]
> One other thing: #416 had a change to `conf/layers.conf` that is not present here, and which has not been sent separately either. What's the status of that?

Thanks for fixing the new patches. There's just the question above that needs to be answered: ^

**Comment by msisov** [comment]
I didn’t have time to send a separate patch with that change because 1) one week was spent on something else 2) another second week was my holidays.

**Comment by rakuco** [comment]
That's OK, what I'm interested in is whether the change is still necessary and, if so, if there are plans to work on it and whether you deliberately removed it from this PR as a result of the feedback in the other PR.

**Comment by msisov** [comment]
That change

**Comment by rakuco** [comment]
OK. In this case you've only answered half of the questions :-) I'm fine with approving this PR as-is, but I'd like some clarity on this first -- can you confirm you intend to fix this separately?

**Comment by r1mikey** [comment]
> OK. In this case you've only answered half of the questions :-) I'm fine with approving this PR as-is, but I'd like some clarity on this first -- can you confirm you intend to fix this separately?

The layerindex question is a bit meta.  I think fixing it here was wrong, as whatever database restriction caused the 25 character limitation in layerindex should be lifted, as it doesn't reflect real-life usage.

**Comment by rakuco** [comment]
> > OK. In this case you've only answered half of the questions :-) I'm fine with approving this PR as-is, but I'd like some clarity on this first -- can you confirm you intend to fix this separately?
> 
> The layerindex question is a bit meta. I think fixing it here was wrong, as whatever database restriction caused the 25 character limitation in layerindex should be lifted, as it doesn't reflect real-life usage.

I agree. My intention with these questions was to make sure this was not forgotten with this PR superseding #416, but I guess just [filing an issue](https://github.com/OSSystems/meta-browser/issues/420) to keep track of it is smarter.

**Review by rakuco — APPROVED** [review]

**Review by otavio — APPROVED** [review]

**Comment by r1mikey** [comment]
One last bit of test signal.   I've run a digital signage video test loop with this PR on Dunfell, x86-64 for 15 hours, and all looks good.

**Pull request merged**